source/iostreams.tex

```diff
@@ -15315,6 +15318,100 @@
 \end{itemdescr}
 
+\rSec3[fs.path.fmtr]{Formatting support}
+
+\rSec4[fs.path.fmtr.general]{Formatting support overview}
+
+\indexlibraryglobal{formatter}%
+\begin{codeblock}
+namespace std {
+  template<class charT> struct formatter<filesystem::path, charT> {
+    constexpr void set_debug_format();
+
+    constexpr typename basic_format_parse_context<charT>::iterator
+      parse(basic_format_parse_context<charT>& ctx);
+
+    template<class FormatContext>
+      typename FormatContext::iterator
+        format(const filesystem::path& path, FormatContext& ctx) const;
+  };
+}
+\end{codeblock}
+
+\begin{ncbnf}
+\nontermdef{path-format-spec}\br
+    \opt{fill-and-align} \opt{width} \opt{\terminal{?}} \opt{\terminal{g}}
+\end{ncbnf}
+where the productions \fmtgrammarterm{fill-and-align} and \fmtgrammarterm{width}
+are described in \ref{format.string}.
+If the \tcode{?} option is used then
+the path is formatted as an escaped string\iref{format.string.escaped}.
+
+\rSec4[fs.path.fmtr.funcs]{Formatting support functions}
+
+\indexlibrarymember{formatter}{set_debug_format}%
+\begin{itemdecl}
+constexpr void set_debug_format();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Modifies the state of the \tcode{formatter} to be as if
+the \fmtgrammarterm{path-format-spec} parsed by the last call to \tcode{parse}
+contained the \tcode{?} option.
+\end{itemdescr}
+
+\indexlibrarymember{formatter}{basic_format_parse_context}%
+\begin{itemdecl}
+constexpr typename basic_format_parse_context<charT>::iterator
+  parse(basic_format_parse_context<charT>& ctx);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Parses the format specifier as a \fmtgrammarterm{path-format-spec} and
+stores the parsed specifiers in \tcode{*this}.
+
+\pnum
+\returns
+An iterator past the end of the \fmtgrammarterm{path-format-spec}.
+\end{itemdescr}
+
+\indexlibrarymember{formatter}{format}%
+\begin{itemdecl}
+template<class FormatContext>
+  typename FormatContext::iterator
+    format(const filesystem::path& p, FormatContext& ctx) const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Let \tcode{s} be \tcode{p.generic_string<filesystem::path::value_type>()}
+if the \tcode{g} option is used,
+otherwise \tcode{p.native()}.
+Writes \tcode{s} into \tcode{ctx.out()},
+adjusted according to the \fmtgrammarterm{path-format-spec}.
+If \tcode{charT} is \keyword{char},
+\tcode{path::value_type} is \keyword{wchar_t}, and
+the literal encoding is UTF-8,
+then the escaped path is transcoded from the native encoding for
+wide character strings to UTF-8 with
+maximal subparts of ill-formed subsequences
+substituted with \ucode{fffd} \uname{replacement character}
+per the Unicode Standard, Chapter 3.9 \ucode{fffd} Substitution in Conversion.
+If \tcode{charT} and \tcode{path::value_type} are the same
+then no transcoding is performed.
+Otherwise, transcoding is
+\impldef{transcoding of the escaped path when \tcode{charT} and \tcode{path::value_type} differ}.
```

Comment:
This is wrong. When the character types differ, transcoding is required whether the path is escaped or not.

```suggestion
\impldef{transcoding of a formatted path when \tcode{charT} and \tcode{path::value_type} differ}.
```

Reply:
We _could_ address this as part of the LWG I'm going to open, but I worry that the index text will be forgotten, because it's not visible in this subclause (only the words "implementation defined" are shown here, the rest is in the annex).

Reply:
please open an issue?

Reply:
We don't open issues against unmerged wording.

Reply:
I'll use your suggestion for now, and I'll send you a reminder when you've opened the issue.

Reply:
This is covered by [LWG 4070](https://cplusplus.github.io/LWG/issue4070) too.